server/src/main/java/org/elasticsearch/index/termvectors/TermVectorsService.java

```diff
@@ -316,7 +316,8 @@ private static Fields generateTermVectorsFromDoc(IndexShard indexShard, TermVect
                 seenFields.add(field.name());
             }
             String[] values = getValues(doc.getFields(field.name()));
-            documentFields.add(new DocumentField(field.name(), Arrays.asList((Object[]) values)));
+            documentFields.add(new DocumentField(field.name(), Arrays.asList((Object[]) values),
+                indexShard.mapperService().isMetadataField(field.name())));
```

Comment:
Just for my own reference: isMetaData information added here also seems to get dropped later when generating the term vectors.